src/main/kotlin/org/bitcoindevkit/karavan/Karavan.kt

```diff
@@ -19,20 +24,81 @@ fun main(args: Array<String>) {
 
 @RestController
 @RequestMapping("/wallet")
-class WalletController() {
-
-    // TODO remove this example
-    @GetMapping("/hello/{name}")
-    fun hello(@PathVariable name: String): String {
-        val descriptor = "wpkh([c258d2e4/84h/1h/0h]tpubDDYkZojQFQjht8Tm4jsS3iuEmKjTiEGjG6KnuFNKKJb5A6ZUCUZKdvLdSDWofKi4ToRCwb9poe1XdqfUnP4jaJjCB2Zwv11ZLgSbnZSNecE/0/*)"
-        val db = DatabaseConfig.Memory("")
-
-        val client =
-            BlockchainConfig.Electrum(
-                ElectrumConfig("ssl://electrum.blockstream.info:60002", null, 5u, null, 10u)
-            )
-        val wallet = OnlineWallet(descriptor, null, Network.TESTNET, db, client)
-        val newAddress = wallet.getNewAddress()
-        return "hello $name, your new address is $newAddress"
+class WalletController(val walletService: WalletService) {
+
+    // Open Wallet by storing the wallet payload in a browser session cookie
+    @PutMapping
+    fun openWallet(response: HttpServletResponse, request: HttpServletRequest, @RequestBody payload: Wallet): String? {
+        // stores payload of type wallet into a cookie on the client's side
+        setCookie(response, payload)
+        // read cookie with key 'wallet' to confirm value
+        return readAllCookies(request)
+    }
+
+    // Close wallet by dropping existing descriptor cookie by setting max-age to 0
+    @DeleteMapping
+    fun closeWallet(request: HttpServletRequest?): String?{
+
+        val cookie = WebUtils.getCookie(request!!, "descriptor")
+        return if (cookie != null) {
+            cookie.maxAge = 0
+            "Wallet is Closed!"
+        } else {
+            "Wallet not found!"
+        }
+    }
+
+    // Get the wallet's balance
+    // Use the browser cookie to get the network and descriptor and then sync
+    // the wallet and get_balance. Will return a balance json with the balance amount.
+    @GetMapping("/getBalance")
```

Comment:
a small thing but per #6 this should be:
```kotlin
@GetMapping("/balance")
```

Reply:
Sorry, I keep messing those up. Will fix now!